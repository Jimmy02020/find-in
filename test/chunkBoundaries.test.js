const fs = require("fs");
const path = require("path");

const find = require("../src/find");

const boundaries = path.join(__dirname, "boundaries.txt");

describe("Testing to match regex on chunk boundaries / issues:2", () => {
  beforeAll(() => {
    const fLen = 64 * 1024 * 3;
    const rLen = 10;
    const i = fLen - rLen - 10;
    const buf = Buffer.alloc(fLen, " ", "utf8");
    buf.fill("B", i - 1, i + rLen + 1, "utf8");
    buf.fill("A", i, i + rLen, "utf8");
    fs.writeFileSync(boundaries, buf);
  });

  afterAll(() => {
    fs.unlinkSync(boundaries);
  });

  it("returns isFound true", async () => {
    const report = await find({ path: boundaries, request: [/BA*B/] });

    expect(report[0].isFound).toBe(true);
    expect(report).toMatchSnapshot();
  });
});
