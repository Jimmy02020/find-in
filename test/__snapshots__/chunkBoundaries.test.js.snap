// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to match regex on chunk boundaries / issues:2 returns isFound true 1`] = `
Array [
  Object {
    "isFound": true,
    "match": Array [
      "BAAAAAAAAAAB",
    ],
    "reg": /BA\\*B/,
  },
]
`;
